WWDC/VideoPlayerViewController.swift

```diff
@@ -177,15 +173,22 @@ final class VideoPlayerViewController: NSViewController {
 
     // MARK: - Player Observation
 
-    override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey: Any]?, context: UnsafeMutableRawPointer?) {
-        guard let keyPath = keyPath else { return }
+    private var playerStatusObserver: NSKeyValueObservation?
+    private var presentationSizeObserver: NSKeyValueObservation?
+    private var currentItemObserver: NSKeyValueObservation?
 
-        if keyPath == #keyPath(AVPlayer.currentItem.presentationSize) {
-            DispatchQueue.main.async(execute: playerItemPresentationSizeDidChange)
-        } else if keyPath == #keyPath(AVPlayer.status) {
-            DispatchQueue.main.async(execute: playerStatusDidChange)
-        } else {
-            super.observeValue(forKeyPath: keyPath, of: object, change: change, context: context)
+    private func setupPlayerObservers() {
```

Comment:
Have you checked if the usage of the new KVO API is fine here? I remember trying to use it before and giving up because AVFoundation objects did not work well with it.

Reply:
I went back to gather some info on this question and found that the function calls were reversed, status change is calling `playerItemPresentationSizeDidChange` and vice versa. Whoops!

The information I gathered is simply this:

```
new status did change
new presentation size did change
new status did change
old status did change
old presentation size did change
old status did change

new presentation size did change
old presentation size did change

new presentation size did change
new presentation size did change
old presentation size did change
old presentation size did change

new presentation size did change
new presentation size did change
new presentation size did change
old presentation size did change
old presentation size did change
old presentation size did change

new presentation size did change
old presentation size did change

new presentation size did change
old presentation size did change
```

All the calls are balanced, though the order of the calls makes me 🤔